// express/lib/router/comp.js



var dom = [];
var na = [];
var na2 =[];
var timer = 1000;
var compName = "data: [";
const https = require('https');
const custID = '009637816073108880163:nfsysoqnztc';
const key = 'AIzaSyBXGkVilmqEN0KSDAaQy1BlVVIA8r4nS6w';



function data(name, domain)  {
  this.name = name;
  this.domain = domain;
}
var finalObject = {};

function printObj(){
  console.log(finalObject);
};

exports.home = function(req, res){
  //var initData = '<%-JSON.stringify(finalObject)%>';
    res.render('index', { title: 'URL Selector', 
      compName: compName, 
      na: na, 
      na2: na2, 
      dom: dom, 
      finalObject: finalObject  })
    res.locals.obj = finalObject;
};

//handler to add company names to array
exports.add_handler = function(req, res){
  if(na.indexOf(req.body.compN) == -1 && req.body.compN.length > 0){
    na.push(req.body.compN);
    console.log(na);
  }
  res.redirect('/');
};

// handler for form submitted from homepage
exports.home_post_handler = function(req, res) {
    
      var temp = "";
      var tempArr= "";

      timer = 1000;
      for(var j = 0; j < na.length; j++){
        
        
        temp = na[j];
        if (temp.length==0){
          res.redirect('/');
        }
        
    
    function got(cName){ https.get('https://www.googleapis.com/customsearch/v1?key='+ key +'&cx=' + custID + '&q=' + temp, function (res){
   
      res.setEncoding("utf8");

      var body = "";

      res.on("data", function(data){
        body += data;
        
       });
  
      res.on("end", function() {
        body = JSON.parse(body);
          console.log(body.searchInformation.searchTime);
          
          timer += body.searchInformation.searchTime * 1000;
        
        if (dom.indexOf(body.items[0].link) == -1){
          dom.push(body.items[0].link);
          na2.push(cName);
          //compName += '{name: ' + temp+ ', domain: ' + body.items[0].link+ '}, ';
          //tempArr += "name: " + cName + " domain:"  + body.items[0].link;
          
          finalObject = new data ( cName, body.items[0].link);
          console.log(finalObject);
          
            };
        //console.log(tempArr);
      //console.log(dom);
    });

    }).on('error', function(err){
      console.error(err)});
      };
    

        got(temp);

      };
   
      
      
      
      // redirect the user to homepage
      setTimeout(function(){
        res.redirect('/');
        
      },timer);
        
      //console.log(compName + 'done');
};


exports.api_handler = function(req, res){
  
  
  if (err){
    throw err;
  }
  res.json(finalObject);


};
