// express/comp.js


//comp.js handles webpage

//creats arrays for names and domains sent to HTML
var dom = []; 
var na = [];
var na2 =[];
//a timer I didn't want to use
var timer = 1000;

const https = require('https');
const custID = '009637816073108880163:nfsysoqnztc';
const key = 'AIzaSyBXGkVilmqEN0KSDAaQy1BlVVIA8r4nS6w';


//object prototype
function data(name, domain)  {
  this.name = name;
  this.domain = domain;
}
var finalObject = {};

function printObj(){
  console.log(finalObject);
};

//handler for home page
exports.home = function(req, res){
 
    res.render('index', { title: 'URL Selector', 
       
      na: na, 
      na2: na2, 
      dom: dom, 
      finalObject: finalObject  })
    res.locals.obj = finalObject;
};

//handler to add company names to array
exports.add_handler = function(req, res){
  if(na.indexOf(req.body.compN) == -1 && req.body.compN.length > 0){
    na.push(req.body.compN);
    console.log(na);
  }
  res.redirect('/');
};

// handler for form submitted from homepage
exports.home_post_handler = function(req, res) {

      var temp = "";
      //var tempArr= ""; unused for now

      //reset timer (hopefully I don't have to use this once I figure out why I had to use it in the first place)
      timer = 1000;
    
      for(var j = 0; j < na.length; j++){
        temp = na[j];
        if (temp.length==0){
          res.redirect('/');
          }
    //get request to google API
    function got(cName){ https.get('https://www.googleapis.com/customsearch/v1?key='+ key +'&cx=' + custID + '&q=' + temp, function (res){
   
      res.setEncoding("utf8");

      var body = "";

      res.on("data", function(data){
        body += data;
        
       });
  
      res.on("end", function() {
        body = JSON.parse(body);
        //log search time
          console.log(body.searchInformation.searchTime);
          timer += body.searchInformation.searchTime * 1000;
        //check if name already used
        if (dom.indexOf(body.items[0].link) == -1){
          //create array for HTML
          dom.push(body.items[0].link);
          na2.push(cName);
          //create obj
          finalObject = new data ( cName, body.items[0].link);
          console.log(finalObject);
          
            };
        });

    }).on('error', function(err){
      console.error(err)});
      };
        got(temp);
      };
     
      // redirect the user to homepage
      setTimeout(function(){
        res.redirect('/');
        },timer);
};


exports.api_handler = function(req, res){
  console.log("get happened");
  res.json(finalObject);
};
